test/e2e/test/helper/yaml.go

```diff
@@ -104,3 +108,212 @@ func (yd *YAMLDecoder) ToBuilders(reader *bufio.Reader, transform BuilderTransfo
 
 	return builders, nil
 }
+
+func (yd *YAMLDecoder) ToObjects(reader *bufio.Reader) ([]runtime.Object, error) {
+	var objects []runtime.Object
+
+	yamlReader := yaml.NewYAMLReader(reader)
+	for {
+		yamlBytes, err := yamlReader.Read()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return nil, fmt.Errorf("failed to read YAML: %w", err)
+		}
+		obj, _, err := yd.decoder.Decode(yamlBytes, nil, nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode YAML: %w", err)
+		}
+
+		objects = append(objects, obj)
+	}
+
+	return objects, nil
+}
+
+// RunFile runs builder workflow for all known resources in a yaml file, all other objects are created before and deleted
```

Comment:
```suggestion
// RunFile runs the builder workflow for all known resources in a yaml file, all other objects are created before and deleted
```
"builder workflow" made me think twice to understand what you mean